web/src/main/java/com/microsoft/applicationinsights/web/internal/httputils/ApplicationInsightsServletExtractor.java

```diff
@@ -0,0 +1,92 @@
+package com.microsoft.applicationinsights.web.internal.httputils;
+
+import org.apache.commons.lang3.Validate;
+import org.apache.http.annotation.Experimental;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Adopter to extract information from {@link HttpServletRequest} and {@link HttpServletResponse}
+ *
+ * Example:
+ * Servlet is mapped as /test%3F/* and the application is deployed under /app.
+ *
+ * http://30thh.loc:8480/app/test%3F/a%3F+b;jsessionid=S%3F+ID?p+1=c+d&p+2=e+f#a
+ *
+ * Method              URL-Decoded Result
+ * ----------------------------------------------------
+ * getMethod()                     GET
+ * getQuery()              no      p+1=c+d&p+2=e+f
+ * getRequestURI()         no      /app/test%3F/a%3F+b;jsessionid=S+ID
+ * getRequestURL()         no      http://30thh.loc:8480/app/test%3F/a%3F+b;jsessionid=S+ID
+ * getScheme()                     http
+ * getHost()                       30thh.loc
+ */
+@Experimental
+public class ApplicationInsightsServletExtractor implements HttpExtractor<HttpServletRequest, HttpServletResponse> {
+
+    private static final String USER_AGENT_HEADER = "User-Agent";
+
+    @Override
+    public String getUrl(HttpServletRequest request) {
+        return request.getRequestURL().toString();
+    }
+
+    @Override
+    public String getMethod(HttpServletRequest request) {
+        return request.getMethod();
+    }
+
+    @Override
+    public String getHost(HttpServletRequest request) {
+        return request.getServerName() + ":" + request.getServerPort();
```

Comment:
This is just to ensure the data consistency with Application Insights previous versions of SDK. Other systems like OpenCensus etc. just return ServerName here.